receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +378,44 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if len(partitionID) > 0 && r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), partitionID)
+		} else {
+			r.writeSuccessResponseWithoutAck(ctx, resp, ld.LogRecordCount())
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) validateAndExtractChannelHeader(req *http.Request) (string, error) {
+	partitionID, ok := req.Header[splunk.HTTPSplunkChannelHeader]
+
+	// check channel header exists
+	if !ok {
+		return "", nil
+	}
+
+	if len(partitionID[0]) == 0 {
```

Comment:
Can you share the expected format of the `partitionID` here? Is it supposed to be a list of UUID strings?

Reply:
Yes, it could be a list of uuids from the request but we'd just take the first one. You can read more about the format here
https://docs.splunk.com/Documentation/Splunk/9.2.1/Data/AboutHECIDXAck#:~:text=curl%20https%3A//mysplunk.com/services/collector%20%2DH%20%22X%2DSplunk%2DRequest%2DChannel%3A%20FE0ECFAD%2D13D5%2D401B%2D847D%2D77833BD77131%22%20%2DH%20%22Authorization%3A%20Splunk%20BD274822%2D96AA%2D4DA6%2D90EC%2D18940FB2414C%22%20%2Dd%20%27%3Cdata%3E%27%20%2Dv